### ISSUE TITLE
No search tool in the knowledge Base displayed on the N4M web portal

### ISSUE THREAD

**Issue by babs1984** [NFDI4Microbiota/nfdi4microbiota-knowledge-base#86]
Therefore searching i.e. for 'ontologies' or other keywords is not possible. Please add a search tool.

**Comment by magelm** [comment]
Additionally:

- [ ] The [search tool on the N4M webpage](https://nfdi4microbiota.de/search) is only selectively searching the webpage:
   - [ ] e.g. searching for "MetaProt" only yields the [news-item](https://nfdi4microbiota.de/news/2023-09-18-metaprot-ws), but not the [subpage](https://nfdi4microbiota.de/usecases/metaprot) or even the [Use Case overview page](https://nfdi4microbiota.de/latest/usecase). 
- [ ] the search bar only offers drop down items to select from (while one is still typing) but hitting "enter" does not yield a preview of the suggested search items

**Comment by Kobert** [comment]
I believe we had a search function before. The text on the Knowledge Base suggests this as well.

https://knowledgebase.nfdi4microbiota.de/ : 
"# How to find a resource
Use the search bar at the top of the left navigation. All the documents have been indexed so a keyword should be sufficient in achieving the look-up for the desired document."

I agree that this should be enabled again.

@babs1984 Is this ticket asking for this, or is it about searching Knowledge Base content from the Web Portal?

**Comment by Kobert** [comment]
> Additionally:
> 
> * [ ]  The [search tool on the N4M webpage](https://nfdi4microbiota.de/search) is only selectively searching the webpage:
>   
>   * [ ]  e.g. searching for "MetaProt" only yields the [news-item](https://nfdi4microbiota.de/news/2023-09-18-metaprot-ws), but not the [subpage](https://nfdi4microbiota.de/usecases/metaprot) or even the [Use Case overview page](https://nfdi4microbiota.de/latest/usecase).
> * [ ]  the search bar only offers drop down items to select from (while one is still typing) but hitting "enter" does not yield a preview of the suggested search items

I would say this is out of scope for the Knowledge Base. We should open a ticket in the Web Portal GitLab instead.

**Comment by magelm** [comment]
> > Additionally:
> > 
> > * [ ]   The [search tool on the N4M webpage](https://nfdi4microbiota.de/search) is only selectively searching the webpage:
> >   
> >   * [ ]   e.g. searching for "MetaProt" only yields the [news-item](https://nfdi4microbiota.de/news/2023-09-18-metaprot-ws), but not the [subpage](https://nfdi4microbiota.de/usecases/metaprot) or even the [Use Case overview page](https://nfdi4microbiota.de/latest/usecase).
> > * [ ]   the search bar only offers drop down items to select from (while one is still typing) but hitting "enter" does not yield a preview of the suggested search items
> 
> I would say this is out of scope for the Knowledge Base. We should open a ticket in the Web Portal GitLab instead.

True, thanks for the suggestion. I added onto the [gitlab issue](https://gitlab.zbmed.de/nfdi4microbiota/nfdi4microbiota_pages/-/issues/102) (one was opened already).

**Comment by magelm** [comment]
Hi @mahdi-robbani , I've hidden the comments that related more to the webportal search bar as off-topic to
focus the discussion on the **search bar for the Knowledge base specifically.**

Maybe we can mimick the setup of Nextflow.io and their documentation page (don't know about the technical background of this setup):
https://nextflow.io/docs/latest/developer/packages.html

PRO:
- you can easily search and preview snippets of your search hits, before clicking on the item
- they somehow dealt with the page restructuring pages problem by (my guess) setting up a toctree and referencing the pages, see here:
https://github.com/nextflow-io/nextflow/blob/232ce9d13e0a98a80447569453b270930a0bb1f6/docs/index.md?plain=1#L38
meaning: we wouldn't need to manually name the docs pages using 00_ anymore and only manually change the index file instead
I think that cpauvert already has a similar set-up in the unmerged PR #154 
- they also don't have the problem with the jumping sidebar (your selection stays at the top of the page for nextflow.io/doc), while the N4M KB sidebar, jumps back to the top and I have to scroll through everything to get to the bottom of the sidebar again). 

**Things to consider:**
Setting up the search function likely depends on decisions made for: https://github.com/NFDI4Microbiota/nfdi4microbiota-knowledge-base/issues/267 
Do we need to take into account that the KB is being moved to a different server? See https://github.com/NFDI4Microbiota/nfdi4microbiota-knowledge-base/issues/282#issuecomment-2687717689

**Comment by mahdi-robbani** [comment]
Search has been added

**Comment by magelm** [comment]
Hi @mahdi-robbani thank you! 
I opened the KB with different browsers and noticed the following:
- full window view: search bar is visible
